demos/defender/defender_demo_json/metrics_collector.h

```diff
@@ -59,6 +59,28 @@ typedef struct Connection
     uint16_t remotePort;
 } Connection_t;
 
+/**
+ * @brief Represents the Cpu Usage statistics obtained from "/proc/uptime".
+ * Refer to Linux manual for "/proc" filesystem for more information.
+ * https://man7.org/linux/man-pages/man5/procfs.5.html
+ */
+typedef struct CpuUsageData
+{
+    uint32_t upTime;   /**< Up-time of system in USER_HZ (1/100th of second) time units. */
+    uint32_t idleTime; /**< Idle time of system in USER_HZ (1/100th of second) time units. */
+} CpuUsageStats_t;
+
+/**
+ * @brief Represents the memory data of total and available memory from "/proc/uptime".
+ * Refer to Linux manual for "/proc" filesystem for more information.
+ * https://man7.org/linux/man-pages/man5/procfs.5.html
+ */
+typedef struct MemoryStats
+{
+    char totalMemory[ 50 ];     /**< Amount of total memory in system (in kB). */
```

Comment:
It will be more efficient to keep `totalMemory` and `availableMemory` as `uint32_t` - thereby using 8 bytes as opposed to 100. 

You can still append "kb" while building the report.